File: src/main/java/arriba/senders/FixTimestampSender.java
package arriba.senders;

import java.io.IOException;

import arriba.common.Sender;
import arriba.fix.inbound.InboundFixMessage;

public final class FixTimestampSender implements Sender<InboundFixMessage> {

    public void send(final InboundFixMessage message) throws IOException {
        final long sendingTime = Long.parseLong((message).getSendingTime());

        final long duration = System.currentTimeMillis() - sendingTime;
        System.out.println(duration);
    }

}

File: src/main/scala/arriba/scala/senders/FixTimestampSender.scala
package arriba.scala.senders

import arriba.fix.messages.FixMessage
import arriba.common.Sender

import java.lang.Long

class FixTimestampSender extends Sender[FixMessage] {
  def send(message: FixMessage) {
    val sendingTime = Long.parseLong((message).getSendingTime)
    val duration = System.currentTimeMillis - sendingTime
    println(duration)
  }
}

